### PR TITLE
Add DOT printer

### PR DESCRIPTION
**Pull request by metanivek**
WIP for hack day with @Ngoguey42

### CODE REVIEW

File: src/repr/type_intf.ml [hunk 1]
```diff
@@ -395,6 +395,9 @@ module type DSL = sig
       native OCaml syntax, so that the result can easily be copy-pasted into an
       OCaml REPL to inspect the value further. *)
 
+  val pp_dot : 'a t pp
```
Comment (Ngoguey42) [inline src/repr/type_intf.ml:398]:
```suggestion
  val pp_DOT : 'a t pp
```

**Comment by Ngoguey42** [comment]
## `Irmin_tezos`'s commit portable
image

## `Irmin_tezos`'s stream proofs
image

## `Irmin_tezos`'s inode`Bin`
image

## `Irmin_tezos`'s tree
image

File: src/repr/type_intf.ml [hunk 2]
```diff
@@ -395,6 +395,9 @@ module type DSL = sig
...
+  (** The DOT language pretty-printer for generics of type {!t}. *)
```
Comment (samoht) [inline src/repr/type_intf.ml:399]: What's DOT?

Comment (metanivek) [reply src/repr/type_intf.ml:399]: https://www.graphviz.org/doc/info/lang.html, language for Graphviz

**Pull request closed without merging**